src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -201,9 +203,21 @@ export class OrganizationIntegrations extends AsyncComponent<
   }
 
   sortIntegrations(integrations: AppOrProviderOrPlugin[]) {
-    return integrations
-      .sort((a, b) => a.name.localeCompare(b.name))
-      .sort((a, b) => this.getInstallValue(b) - this.getInstallValue(a));
+    const sortByName = (a, b) => a.name.localeCompare(b.name);
+    const sortByPopularity = (a, b) => {
+      const weightA = Number.isInteger(weights[a.slug]) ? weights[a.slug] : 1;
+      const weightB = Number.isInteger(weights[b.slug]) ? weights[b.slug] : 1;
+      return weightA > weightB ? -1 : weightB > weightA ? 1 : 0;
+    };
+    const sortByInstalled = (a, b) => this.getInstallValue(b) - this.getInstallValue(a);
```

Comment:
Just wanted to point out that the implementation of `this.getInstallValue` has a nested `if` logic which should really be refactored.

Reply:
@leedongwei not sure if there is any value in factoring out the `if` logic. It's not going to be reused anywhere else.

Reply:
You misunderstood me. The method doesn't need to be split up, it's fine on its own chunk of code. However, it needs to be rewritten for better readability (i.e. favour returning as early as possible to avoid nested if-statements, remove `if (..) { return } else { .. }`)

Reply:
@leedongwei can't refactor too much here b/c we can only call `this.getAppInstall` when `integration` is a SentryApp type.